### ISSUE TITLE
Arrow function should not return assignement.

### ISSUE THREAD

**Issue by slim-hmidi** [eslint/eslint#10633]
**Tell us about your environment**

* **ESLint Version:** 4.17.0
* **Node Version:** 8.11.3
* **npm Version:** 5.6.0

**What parser (default, Babel-ESLint, etc.) are you using?**
**eslint-config-airbnb-base**: 12.1.0
**eslint-plugin-import**: 2.8.0

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
    "parserOptions": {
        "ecmaVersion": 2017
    },
    "env": {
        "node": true
    },
    "extends": "airbnb-base"
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
Object.keys(includedDayStock).map(key => accumulator[key] = includedDayStock[key]);
```

```bash

```

**What did you expect to happen?**
No error message.

**What actually happened? Please include the actual, raw output from ESLint.**
I got eslint error `Arrow function should not return assignement `. I found a similar issue #5150 but the answers mentioned did not resolve the problem.

**Comment by not-an-aardvark** [comment]
Hi,

This error is being reported because your arrow function is returning an assignment expression.

To fix the issue, you can do something like this:

```js
Object.keys(includedDayStock).map(key => {
  accumulator[key] = includedDayStock[key];
});
```

Alternatively, for the specific case of copying the keys of an object, you could do this:

```js
Object.assign(accumulator, includedDayStock);
```
